app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -316,6 +320,54 @@ protected function getRatesDebuggingData($rates) {
         return $rateDebuggingData;
     }
 
+    protected function isDiscountRoundingDeltaError($transaction, $quote) {
+        $boltDiscountAmount = $this->getBoltDiscountAmount($transaction);
+        $quoteDiscountAmount = $quote->getShippingAddress()->getDiscountAmount();
+
+        return $this->isOnePennyRoundingError($boltDiscountAmount, $quoteDiscountAmount);
+    }
+
+    protected function getBoltDiscountAmount($transaction) {
+        $boltDiscountAmount = 0;
+
+        if(isset($transaction->order->cart->discounts)) {
+            foreach($transaction->order->cart->discounts as $discount) {
+                if(isset($discount->amount->amount) && is_numeric($discount->amount->amount) && $discount->amount->amount > 0) {
+                    $boltDiscountAmount -= $discount->amount->amount / 100;
+                }
+            }
+        }
+
+        return $boltDiscountAmount;
+    }
+
+    protected function isOnePennyRoundingError($boltDiscountAmount, $quoteDiscountAmount) {
+        if($boltDiscountAmount != $quoteDiscountAmount) {
+            $difference = round(abs($boltDiscountAmount - $quoteDiscountAmount), 2);
```

Comment:
I prefer we don't assume sign of boltdiscountamount of quotediscountamount so `round(abs(A) - abs(B), 2)`

Reply:
I considered this. But what about the case where the Bolt discount is positive and the Magento quote is negative (we're currently allowing this with shipping rate discounts). So for example, let's say we have a Bolt discount amount of $8.01 and a Magento discount amount of -$8.00. If we do round(abs(A) - abs(B), 2) we will end up with $8.01 - $8.00 = $0.01 which will incorrectly flag this error. Instead it should be $8.01 - (-$8.00) = $16.01 which is accurate and won't flag this as a rounding error.